realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -37,26 +36,27 @@
 
     private static final long nativeFinalizerPtr = nativeGetFinalizerPtr();
 
-    // See documentation in that NativeContext for an explanation of how this is used
-    @SuppressWarnings("unused")
-    private final NativeContext context;
-
     private final Table table;
     private final long nativePtr;
 
-    // All actions (find(), findAll(), sum(), etc.) must call validateQuery() before performing
-    // the actual action. The other methods must set queryValidated to false in order to enforce
-    // the first action to validate the syntax of the query.
+    private final MixedNativeFunctionsImpl mixedNativeFunctions = new MixedNativeFunctionsImpl();
+
     private boolean queryValidated = true;
 
-    // TODO: Can we protect this?
-    public TableQuery(NativeContext context, Table table, long nativeQueryPtr) {
+    private static String escapeFieldName(String fieldName) {
```

Comment:
Escaping field names on non-`rawPredicate` means a discrepancy between typed and raw queries. Maybe not that in the CHANGELOG as know bug (assuming it is a bug in the query parser causing this).

Reply:
Not sure it is a bug as such, just something we need to document for using the string-based queries.

Reply:
I would expect the parser to do any escaping on field names like it is done on arguments.

Reply:
I'm not sure it is possible to encode that in the query parser tokenizer

Reply:
We could probably support `"name with spaces"` but that would just make `"` the magic character that needs escaping. 

The only place spaces could occur is through the dynamic Realm as most languages wouldn't allow it, so just documenting seems like the most sane approach to me.